uwblueprint/sistema
# Implement input form dropdown functionality

## Notion ticket link



[Dropdowns in Calendar Input Form - Functionality](https://www.notion.so/uwblueprintexecs/Dropdowns-in-Calendar-Input-Form-Functionality-12f10f3fb1dc8001af6bc21e9764306f)



## Implementation description
- Created API route to fetch Subject and Location data for dropdown options
- Created the dropdown component
- Implemented de-selection of already-selected option when clicked again
- Integrated the dropdown component into InputForm's Subject and Location fields



## Steps to test

1. Visit http://localhost:3000/calendar
2. Open the input form via the "declare absence" button or a date block click
3. Verify Subject dropdown selection when an option is clicked
4. Verify Subject dropdown de-selection when an already-selected option is clicked again
5. Verify Location dropdown selection when an option is clicked
4. Verify Location dropdown de-selection when an already-selected option is clicked again
6. Input all fields but Subject dropdown - verify validation alert
7. Input all fields but Location dropdown - verify validation alert
8. Input all fields including dropdown - verify form submission via the "declare absence" button
9. Verify absence is populated in the calendar and database with the correct Subject and Location



## What should reviewers focus on?

- Intuitive and user-friendly dropdown experience (selection, de-selection, navigation)
- Proper reflection of declared absence in the calendar and database
- General code style/best practices

## Checklist
